server/routes/soonBusStop.js

```diff
@@ -0,0 +1,127 @@
+import { calcCWorCCW } from "./direction.js"
+const busStops = require('./bus-stops.json')
+const defaultDatabase = require('./firebase.js')
+
+let bussesRef;
+let busID = [];
+let previousLocationArray = [];
+let stops_arr_CW = [];
+let stops_arr_CCW = [];
+let lat1 = 0  // Current longitude
+let lon1 = 0  // Current latitude
+
+// updates next 3 bus stops for every bus
+export function nextBusStops() {
+  bussesRef = defaultDatabase.collection('busses');
+
+  // Push all bus ID's into array
+  defaultDatabase.collection("cities").get().then(function(querySnapshot) {
+    querySnapshot.forEach(function(doc) {
+        busID.push(doc.id);
+    });
+  });
+
+  // For each bus stop id, pass the id and store upcoming bus stops into array
+  busID.forEach(soonBusStop);
+
+  // Update the database for each bus stops in the array
+  dbUpdate(stops_arr_CW, "CW");
+  dbUpdate(stops_arr_CCW, "CCW");
+
+  // Empty the arrays
+  busID = [];
+  previousLocationArray = [];
+  stops_arr_CW = [];
```

Comment:
Also might be easier to keep all the bus stops as a map instead of array, where the key is the bus stop, and the value is either true or false for soon.